pyminio/main.py

```diff
@@ -391,25 +435,22 @@ def cp(self, from_path: str, to_path: str, recursive: bool = False):
             recursive: copy content recursively.
         """
         from_match = Match(from_path)
-
-        # TODO: implement copy recursive (Task #12)
-        # if from_match.is_dir() and not recursive:
-        #     raise ValueError("copying a directory must be done recursively")
+        to_match = self._get_destination(from_path, to_path)
 
         if from_match.is_dir():
-            raise NotImplementedError("cannot copy folder, currently "
-                                      "unsupported")
+            if recursive:
+                self.copy_recursively(from_match.path, to_match.path)
+                return
```

Comment:
return self.copy_recursively
in case there will be a return value

Reply:
also consider returning self to support chaining:
self.cp().mv().rm() for example

Reply:
d&d